tests/unit_tests/scheduler/test_scheduler.py

```diff
@@ -444,3 +445,14 @@ async def wait():
         for start, next_start in zip(run_start_times[:-1], run_start_times[1:])
     ]
     assert min(deltas) >= submit_sleep * 0.8
+
+
+@pytest.mark.timeout(5)
+async def test_exceptions_are_propagated(realization):
+    driver = Mock()
+    driver.poll = lambda: 1 / 0
```

Comment:
do the same with _process_event_queue and _publisher?

Reply:
Yes